Base/Enums/EnumUtils.cs

```diff
@@ -0,0 +1,37 @@
+﻿/*
+ * This is a hack, and should go away!
+ *
+ * discovered here: https://stackoverflow.com/a/28527552
+ */
+
+using System.IO.BACnet.Serialize;
+
+namespace System.IO.BACnet
+{
+    public abstract class EnumClassUtils<TClass> where TClass : class
+    {
+
+        public static int DecodeEnumerated<TEnum>(byte[] buffer, int offset, uint lenValue, out TEnum value)
+        {
+            var len = ASN1.decode_unsigned(buffer, offset, lenValue, out var rawValue);
+            value = (TEnum)(dynamic)rawValue;
+            return len;
+        }
+
+        public static int DecodeContextEnumerated<TEnum>(byte[] buffer, int offset, byte tagNumber, out TEnum value)
```

Comment:
I made this comment in previous PR. You forgot to put constraint where TEnum : struct, TClass